source/PlayerInfo.cpp

```diff
@@ -129,6 +129,65 @@ namespace {
 
 
 
+PlayerInfo::ScheduledEvent::ScheduledEvent(const DataNode &node, const ConditionsStore *conditions)
+{
+	GameEvent nodeEvent(node, conditions);
+
+	string eventName;
+	if(!nodeEvent.Name().empty())
+		eventName = nodeEvent.Name();
+	else
+	{
+		// Old save files may contain unnamed events. In that case, the event's name can be found by
+		// looking at the relevant conditions in the event's conditions assignment.
+		set<string> conditions = nodeEvent.Conditions().RelevantConditions();
+		erase_if(conditions, [](const string &name) { return name.find("event: ") == string::npos; });
+		// Unless the save file was manually altered, there should be an event condition present.
+		// If there are multiple event conditions present, then the first one should be the condition
+		// for this event, as assignments are saved and loaded in the order they're created, and
+		// the event condition is the first assignment added to each event.
+		if(!conditions.empty())
+			eventName = conditions.begin()->substr(strlen("event: "));
+	}
+	if(!eventName.empty())
+		event = ExclusiveItem<GameEvent>(GameData::Events().Get(eventName));
+	else
+	{
+		// Fall back onto saving the full definition if we somehow didn't find a name.
+		// This event will still properly apply its changes, but they may be applied
+		// out of order.
+		node.PrintTrace("Warning: Could not determine name of unnamed event.");
+		event = ExclusiveItem<GameEvent>(std::move(nodeEvent));
+	}
+
+	date = nodeEvent.GetDate();
```

Comment:
`nodeEvent.GetDate()` might return something unexpected after the std::move on line 160.